src/statemachine.rs

```diff
@@ -552,12 +568,37 @@ impl StateMachine {
                         }
                     };
 
-                    match makecred.do_pre_flight(&mut dev) {
-                        Ok(()) => { /* Continue with the request */ }
-                        Err(e) => {
-                            warn!("error happened: {e}");
-                            callback.call(Err(e));
-                            break;
+                    // Do "pre-flight": Filter the exclude-list
+                    if dev.get_authenticator_info().is_some() {
+                        makecred.exclude_list = unwrap_result!(
```

Comment:
I don't think `unwrap_result` improves readability here. Better to explicitly handle the cases, especially since we can return early with `CredentialExcluded` if the resulting `exclude_list` is non-empty.

Reply:
I'm not 100% sure why yet, but Chromium is explicitly requiring a touch by the user in both error-cases (`CredentialExcluded` here and `NoCredentials` for GetAssertion). For GetAssertion, it even creates a dummy request, just to get the user-interaction. I'm still skimming the spec to find out, if it is mentioned there somewhere.

Either way, we are currently inconsistent (user-interaction for MC, no user-interaction for GE). We should decide which way to go, and do it consistently. Both ways are easily possible.

Reply:
Ah, [there](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#op-makecred-step-if-excludeList) it is for MC:
> Note: A user presence test is required for CTAP2 authenticators, before the RP is told that the authenticator is already registered, to behave similarly to CTAP1/U2F authenticators.

I currently can't find a strict requirement for GA, though. But we did this in the U2F-case as well: https://github.com/mozilla/authenticator-rs/blob/main/src/statemachine.rs#L229-L238

Reply:
Ah, right, good point!